src/alert.sh

```diff
@@ -0,0 +1,22 @@
+# NOTE: src/commons.sh must be included before this file
+
+# Alerts command completion to the user. First argument is the alert code,
+# and second is the command string to be printed in visual notification.
+function alert_completion()
+{
+
+  local COMMAND=$1
+  local ALERT_OPT=$2
+
+  if [[ $# -gt 1 && "$2" =~ ^--alert= ]]; then
+    local opts="$(echo $2 | sed s/--alert=// )"
+
+    grep -o . <<< "$opts" | while read option;  do
+      if [ "$option" == "v" ]; then
+        eval "${configurations[visual_alert_command]}"
```

Comment:
We should avoid eval command in the Kw, especially here. Why? The user can set the value loaded by kw, and here live the dangerous of this operation. A malicious user could set a command that "rm" the whole home for example. For extra information take a look at:

https://unix.stackexchange.com/questions/278427/why-and-when-should-eval-use-be-avoided-in-shell-scripts

Anyway, we can replace the eval command by simply call the player. Just remember to add the check if the program is available or not.

Reply:
I didn't understand the problem, can you explain it to me a little further? I mean, if we change `eval $COMMAND_SET_BY_USER` to `$COMMAND_SET_BY_USER`, won't the command, which can be an `rm` as you said, be executed anyway?

Reply:
Good point here!

I'm worried about this kind of operation because I have plans to package kw for Debian and Arch. If we add kw as a real distro package (e.g., apt install kw), we have to take extra attention to not create a problem for users. In this sense, I want to avoid eval and other sorts of operations that can accidentally/maliciously compromise the system.

Now, I think that could be better if we only support paplay and send-notify. In the future, if someone asks we can add other tools.

Reply:
Ok, I understand your security concern here. But the "malicious user" would be setting a command to run on his own machine. So he can't do harm to anyone but himself...

Is your concern more related to a situation where a ks user is downloading something from the internet that maliciously comes with a "kworkflow.config" file, set with a "rm" command? In this case, the user might run a "kw something" and execute arbitrary code not desired by him. But to try preventing this situation seems a little bit overkill to me. What do you thing? Should I go and remove the customization anyway?

Reply:
Ok, you're right. It's just paranoid from my side. Just ignore it, and go ahead.